src/kem/kyber768.rs

```diff
@@ -55,15 +59,18 @@ pub fn encapsulate(
 ) -> Result<(Kyber768Ciphertext, Kyber768SharedSecret), BadRejectionSamplingRandomnessError> {
     let randomness_hashed = H(&randomness);
 
-    let to_hash: [u8; 2 * H_DIGEST_SIZE] = randomness_hashed.push(&H(&public_key));
+    let mut to_hash: [u8; 2 * H_DIGEST_SIZE] = randomness_hashed.as_ref().into_padded_array();
```

Comment:
Do these changes make a performance difference? Or what is the motivation for the change?

Reply:
No I don't believe they do,  I found the `UpdatingArray2` traits and methods a bit confusing and I decided to remove them to keep just one. I might've been a little too quick to do so.

Reply:
Simplifying the code is a good reason as well 🙂